project1/constantblackscholesprocess.cpp

```diff
@@ -1,3 +1,31 @@
 
 #include "constantblackscholesprocess.hpp"
+#include <iostream>
 
+namespace QuantLib {
+
+	ConstantBlackScholesProcess::ConstantBlackScholesProcess(double S_, double r_, double sigma_, double q_) : S(S_), r(r_), sigma(sigma_), q(q_) {
```

Comment:
Found it.  The default constructor for StochasticProcess1D, from which you inherit, doesn't initialize the shared_ptr<discretization> which it contains.  You can pass an instance here:
```suggestion
	ConstantBlackScholesProcess::ConstantBlackScholesProcess(double S_, double r_, double sigma_, double q_)
	: StochasticProcess1D(ext::shared_ptr<discretization>(new EulerDiscretization)),
	  S(S_), r(r_), sigma(sigma_), q(q_) {
```
You'll need to include `<ql/processes/eulerdiscretization.hpp>`.

With this change, your program seems to work.  Perform your own checks and measurements; then you can start thinking about the report.